torch/lib/c10d/ProcessGroupMPI.hpp

```diff
@@ -77,24 +81,31 @@ class ProcessGroupMPI : public ProcessGroup {
   class WorkMPI : public ProcessGroup::Work {
    public:
     WorkMPI(
+        const std::shared_ptr<std::vector<at::Tensor>> outputs,
         const char* profilingTitle = nullptr,
         const c10::optional<std::vector<at::Tensor>>& inputTensors =
             c10::nullopt)
         : ProcessGroup::Work(
               -1,
               OpType::UNKNOWN,
               profilingTitle,
-              inputTensors) {}
+              inputTensors),
+          outputs_(outputs) {}
```

Comment:
curious, why `const std::shared_ptr` + copy instead of `std::shared_ptr` + move?